kubernetes/kubernetes.libsonnet

```diff
@@ -36,6 +36,34 @@ local objectMetadata(object, config) =
     } + config.podAnnotations
   );
 
+// Function to create volumes in deployment
+local volumeService(volumeSpec) = [
+  local volume = k.core.v1.volume;
+  local name = volume.withName(vol.name);
+  volume.withName(vol.name)
+  + (if std.objectHas(vol, 'awsElasticBlockStore') then volume.awsElasticBlockStore.withFsType(vol.awsElasticBlockStore.fsType) else {})
+  + (if std.objectHas(vol, 'awsElasticBlockStore') then volume.awsElasticBlockStore.withVolumeID(vol.awsElasticBlockStore.volumeID) else {})
+  + (if std.objectHas(vol, 'awsElasticBlockStore') then volume.awsElasticBlockStore.withPartition(vol.awsElasticBlockStore.partition) else {})
+
+  + (if std.objectHas(vol, 'configMap') then volume.fromConfigMap(name, configMapName=vol.configMap.name) else {})
+
+  + (if std.objectHas(vol, 'secret') then volume.fromSecret(name, secretName=vol.secret.name) else {})
+
+  
+  for vol in volumeSpec
+];
+
+// Function to create volumeMounts in container
+local containerVolume(volumeSpec) = [
```

Comment:
I'd like it if all arguments that are arrays be in plural. So here `volumeSpec` is an array but the name suggests that it's a single spec.